player/magazine.gd

```diff
@@ -1,15 +1,18 @@
 class_name PlayerMagazine
 extends Magazine
 
+var _base_ammo: int
 
-func _init(ammo: int, max_ammo: int) -> void:
-	super(ammo, max_ammo)
+
+func _init(base_ammo: int) -> void:
+	super(base_ammo, base_ammo)
+	_base_ammo = base_ammo
```

Comment:
The magazine upgrade doesn't persist changing weapons, need to solve this outside each weapon.